src/api.js

```diff
@@ -2087,6 +2087,79 @@ api.declare({
   return res.reply(result);
 });
 
+/** Get a worker-type from a provisioner */
+api.declare({
+  method:     'get',
+  route:      '/provisioners/:provisionerId/worker-types/:workerType',
+  name:       'getWorkerType',
+  stability:  API.stability.experimental,
+  output:     'workertype-response.json#',
+  title:      'Get a worker-type',
+  description: [
+    'Get a worker-type from a provisioner.',
+  ].join('\n'),
+}, async function(req, res) {
+  const {provisionerId, workerType} = req.params;
+
+  const wType = await this.WorkerType.scan({
+    provisionerId,
+    workerType,
+    expires: Entity.op.greaterThan(new Date()),
+  });
+
+  const result = wType.entries.length ? wType.entries[0].json() : {};
+
+  return wType.entries.length ?
```

Comment:
Rather than two `?` expressions, a conditional with `return res.reportError(..)` would be simpler.